src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -545,17 +546,48 @@ public static String removeNonAscii(String text) {
      * with a copy of the specified delimiter.
      */
     public static String join(String delimiter, String... elements) {
-        StringBuffer result = new StringBuffer();
+        if (elements == null || elements.length == 0) {
+            return "";
+        }
```

Comment:
Looks like this method interprets null as `","` and `empty list` respectively, for first and second parameters. It is risky to treat null as a legit input because null can be a result of a programming error. Why not force the caller to use `","` and `empty list` if that's the intention of the caller? If we must use null in this way, we must at least update the header comment of this method to explain this additional behavior.